python/hidet/graph/ops/definitions/normalize/norm_f16.py

```diff
@@ -0,0 +1,272 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import List
+from hidet.ir import IRModule, dtypes
+from hidet.ir.primitives import active_mask, shfl_down_sync
+from hidet.lang import spatial, repeat, view, cast
+from hidet.lang import data_type, TensorType, i32, f16, attrs, tensor
+from hidet.lang.cuda import blockIdx, threadIdx, register_tensor, syncthreads
+from hidet.graph.ops.definitions.utils import Operator, Tensor, normalize_dim
+from hidet.graph.ops.definitions.utils import input_like
+from hidet.utils import prod
+from hidet.ir import primitives as prim
+from .norm import NormalizeTask
+
+
+class NormalizeF16Task(NormalizeTask):
```

Comment:
I am not sure which one is better: implement a fp16 version seperately vs. implement the fp16 & fp32 version in the same template.

Reply:
with 2xf16 it is probably better to keep them seperate. I will add it.

Otherwise most of the logic is the same and we can basically template on <input dtype, accumulate dtype>